bundles/org.openhab.binding.liquidcheck/src/main/java/org/openhab/binding/liquidcheck/internal/LiquidCheckHandler.java

```diff
@@ -0,0 +1,173 @@
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link LiquidCheckHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Goerentz - Initial contribution
+ */
+@NonNullByDefault
+public class LiquidCheckHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(LiquidCheckHandler.class);
+    private final HttpClient httpClient;
+
+    private Map<String, String> oldProps = new HashMap<>();
+
+    private LiquidCheckConfiguration config = new LiquidCheckConfiguration();
+    private LiquidCheckHttpClient client = new LiquidCheckHttpClient(config, new HttpClient());
+
+    private @Nullable ScheduledFuture<?> polling;
+
+    public LiquidCheckHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(MEASURE_CHANNEL)) {
+            if (command instanceof OnOffType) {
+                try {
+                    if (client.isConnected()) {
+                        String response = client.measureCommand();
+                        CommData commandResponse = new Gson().fromJson(response, CommData.class);
+                        if (commandResponse != null) {
+                            if (!"success".equals(commandResponse.context.status)) {
+                                logger.warn("Starting the measurement was not successful!");
+                            }
+                        } else {
+                            logger.debug("The object commandResponse is null!");
+                        }
+                    }
+                } catch (TimeoutException | ExecutionException | JsonSyntaxException e) {
+                    logger.warn("This went wrong in handleCommand: {}", e.getMessage());
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+                updateState(channelUID, OnOffType.OFF);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(LiquidCheckConfiguration.class);
+        oldProps = thing.getProperties();
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            this.client = new LiquidCheckHttpClient(config, httpClient);
+            PollingForData pollingRunnable = new PollingForData(this.client);
+            polling = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, config.refreshInterval, TimeUnit.SECONDS);
+        });
```

Comment:
Simplification possible:
```suggestion
        this.client = new LiquidCheckHttpClient(config, httpClient);
        PollingForData pollingRunnable = new PollingForData(this.client);
        polling = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, config.refreshInterval, TimeUnit.SECONDS);
```